### DISCUSSION TITLE
Audit service over rides fields in keyvaluemap

### DISCUSSION THREAD

**Discussion by manimaarans** [egovernments/Digit-Core#629] [Bugs]
Audit service overrides the key value pair if sanitized jsonpath are same for more than one entry

sanitized jsonpath implementation: https://github.com/egovernments/Digit-Core/blob/347bafe2806a43e928bfd6473c96ab4ee0431925/core-services/audit-service/src/main/java/org/egov/auditservice/persisterauditclient/PersisterAuditClientService.java#L428

keyvalueMap: https://github.com/egovernments/Digit-Core/blob/347bafe2806a43e928bfd6473c96ab4ee0431925/core-services/audit-service/src/main/java/org/egov/auditservice/persisterauditclient/PersisterAuditClientService.java#L134

persister config where this is happening : in the following example litigant id is replaced by case id 
this litigant id : https://github.com/pucardotorg/kerala-configs/blob/6d20f5a33aff4857c2cd414ca68fc700035b9a85/egov-persister/case-persister.yml#L270
 case id : https://github.com/pucardotorg/kerala-configs/blob/6d20f5a33aff4857c2cd414ca68fc700035b9a85/egov-persister/case-persister.yml#L277

this makes it difficult to go though audit trail as we are expecting to track by litigant id to identify the unique records , but  caseid present instead of it.
any suggestions on this ?

**Comment by manimaarans** [comment]
can we get the latest image build  for this https://github.com/egovernments/Digit-Core/tree/audit-details-extractSanitizedFieldNameFromJsonPath?

also https://github.com/pucardotorg/kerala-configs/blob/6d20f5a33aff4857c2cd414ca68fc700035b9a85/egov-persister/case-persister.yml#L313 for this with above implementation , the key would be "cases.representatives[*][?({id} in @.representing[*].id)].id" . it is not blocker as there wont be any overrides and we can uniquely identify them

**Comment by Rishabh-egov** [comment]
Hey @manimaarans please find the latest build: `egovio/audit-service-db:v2.9.0-e8539a9372-15`
For the second : There isn't a way for the service to know what key to use in case of duplicates , to make the solution encompassing all scenario we have to go with the full path as the key. 
If the requirement is for custom key then please directly call signed audit API's with the custom keys.

**Reply by manimaarans** [reply]
> Hey @manimaarans I have shared the build with Atul - check with him. For the second : There isn't a way for the service to know what key to use in case of duplicates , to make the solution encompassing all scenario we have to go with the full path as the key. If the requirement is for custom key then please directly call signed audit API's with the custom keys.

Thanks for the response @Rishabh-egov

**Reply by Rishabh-egov** [reply]
Hey @manimaarans updated the comment with the latest bulid.
